### DISCUSSION TITLE
`supportedArchitectures` setting

### DISCUSSION THREAD

**Discussion by alicewriteswrongs** [yarnpkg/berry#3855] [Q&A]
Hello! I'm using the `supportedArchitectures` setting in `.yarnrc.yml` in an attempt to have yarn install packages from `@swc` for both `darwin` and `linux`, and for both `x64` and `arm64`. I'm also using `nodeLinker: "node-modules"` because swc doesn't yet support pnp.

Here is my `yarnrc.yml`:

```
yarnPath: ".yarn/releases/yarn-3.1.0.cjs"
nodeLinker: "node-modules"
supportedArchitectures:
  os: [linux, darwin]
  cpu: [x64, arm64]
```

The thing I'm running into is that when I do a `yarn install` it still only installs the packages for the current platform. Is that the intended behavior here? Is there any way to get yarn to always install the architectures I've listed here, regardless of what platform of the current host is?

I have a rather specific and slightly funny use-case for this. We run a dev build of our React app in a docker container, so when that container boots it runs `yarn install`, installing `linux` and `x64` packages. However, I'm running docker on a macOS host machine, and I like being able to do things like run tests from the host without needing to run everything within the docker container. The `node_modules` directory is shared between the host and the container, so whichever one runs a yarn install decides which architecture is installed. I'd like to have them both installed all the time so that I don't need to remember whether I ran `yarn install` on the host machine or in the container, and so everything can 'just work' :)

lmk if anything here doesn't make sense! I'm just hoping that I have something misconfigured :)

**Comment by yuriy-yarosh** [comment]
@alicewriteswrongs Nah... it's broken for both `eslint` and `swc` now. 

`yarn set version 3.7.0` fixes things...
Mentioned here https://github.com/evanw/esbuild/issues/1819
https://github.com/yarnpkg/berry/issues/4615
